### PR TITLE
Code Refactoring

### PR DESCRIPTION
**Pull request by anakin001**
refactoring networkState->appstate,remove setFee & env, fix hashlock, remove unused imports, remove unecessary parameters

### CODE REVIEW

File: src/modules/services/submodule/aggregateTransactions/views/ViewServicesAggregateTransactionsComplete.vue [hunk 1]
```diff
@@ -551,7 +552,7 @@ export default {
 
       let transactionBuilder = new BuildTransactions(NetworkType)
       
-      if (ChainUtils.getNetworkType(networkState.currentNetworkProfile.network.type) === NetworkType.PRIVATE || ChainUtils.getNetworkType(networkState.currentNetworkProfile.network.type) === NetworkType.PRIVATE_TEST) {
+      if (ChainUtils.getNetworkType(AppState.networkType) === NetworkType.PRIVATE || ChainUtils.getNetworkType(AppState.networkType) === NetworkType.PRIVATE_TEST) {
         transactionBuilder.setFeeStrategy(FeeCalculationStrategy.ZeroFeeCalculationStrategy) ;
         //FeeCalculationStrategy.ZeroFeeCalculationStrategy
       }
```
Comment (sleepyOwl14) [inline src/modules/services/submodule/aggregateTransactions/views/ViewServicesAggregateTransactionsComplete.vue:558]: this file not used right ?

File: src/util/swapUtils.ts [hunk 2]
```diff
@@ -747,7 +748,7 @@ export class SwapUtils {
     const accountDetails = walletState.currentLoggedInWallet.accounts.find((account) => account.address == accAddress.plain());
     const encryptedPassword = WalletUtils.createPassword(walletPassword);
     let privateKey = WalletUtils.decryptPrivateKey(encryptedPassword, accountDetails.encrypted, accountDetails.iv);
-    const account = Account.createFromPrivateKey(privateKey, ChainUtils.getNetworkType(networkState.currentNetworkProfile.network.type));
+    const account = Account.createFromPrivateKey(privateKey, ChainUtils.getNetworkType(AppState.networkType));
```
Comment (sleepyOwl14) [inline src/util/swapUtils.ts:751]: just use AppState.networkType directly, no need to use ChainUtils.getNetworkType

**Review by sleepyOwl14 — CHANGES_REQUESTED** [review]
missing AppState import on :
- SelectAccountTypeModal.vue
- ViewAccountCreate.vue
- ViewAccountCreatePrivateKey.vue

please check again

File: src/util/accountUtils.ts [hunk 3]
```diff
@@ -222,40 +205,14 @@ const getLinkNamespaceToAddressTransactionFee = (isMultisig :boolean,namespaceAd
...
-  AppState.chainAPI.transactionAPI.announce(lockHashSigned);
-
-  // ListenerStateUtils.addAutoAnnounceSignedTransaction(hashLockAutoAnnounceSignedTx);
-  ListenerStateUtils.addAutoAnnounceSignedTransaction(autoAnnounceSignedTx);
-
-  AppState.isPendingTxnAnnounce = true;
-}
-
-
 const linkNamespaceToAddress = (isMultisig :boolean,cosigners: [], multisigAccount: WalletAccount | OtherAccount, walletPassword: string, namespaceID: string, linkType: string, namespaceAddress: string) :SignedTransaction=> {
```
Comment (sleepyOwl14) [inline src/util/accountUtils.ts:208]: should pass in only 1 selected cosigner to sign, if multisig

Comment (sleepyOwl14) [reply src/util/accountUtils.ts:208]: and the data type is missing

Comment (anakin001) [reply src/util/accountUtils.ts:208]: delegate and linkNamespaceToAddress doesnt have the UI for selecting cosigner, might need to add it

Comment (anakin001) [reply src/util/accountUtils.ts:208]: as well as multisig, only transfer asset and namespace can select

File: src/util/accountUtils.ts [hunk 4]
```diff
@@ -267,67 +224,47 @@ const linkNamespaceToAddress = (isMultisig :boolean,cosigners: [], multisigAccou
...
-    const lockFundsTransaction = txBuilder.hashLock(  
-      new Mosaic(new NamespaceId(nativeTokenNamespace), UInt64.fromUint(lockingAtomicFee)),
-      UInt64.fromUint(1000),
-      signedAggregateBondedTransaction,
-    )
-    const lockFundsTransactionSigned = firstSigner.sign(lockFundsTransaction , Hash);
-    signedTransaction = signedAggregateBondedTransaction
-    multiSigAnnouce(signedAggregateBondedTransaction,lockFundsTransactionSigned)
+    const lockFundsTransaction = TransactionUtils.lockFundTx(signedAggregateBondedTransaction)
+    const lockFundsTransactionSigned = firstSigner.sign(lockFundsTransaction, Hash);
```
Comment (sleepyOwl14) [inline src/util/accountUtils.ts:233]: please use proper name for generation hash, genHash in short. 
btw, the way to use networkType and generationHash is wrong. It is in a fixed value

File: src/util/transactionUtils.ts [hunk 5]
```diff
@@ -59,22 +59,18 @@ import {
...
-import { faBreadSlice } from "@fortawesome/free-solid-svg-icons";
 import { AppState } from "@/state/appState";
+import { ChainConfigUtils } from "./chainConfigUtils";
+import { ListenerStateUtils } from "@/state/utils/listenerStateUtils";
+import { AnnounceType, AutoAnnounceSignedTransaction, HashAnnounceBlock } from "@/state/listenerState";
+import { networkState } from "@/state/networkState";
 
-const networkAPIEndpoint = computed(() => ChainUtils.buildAPIEndpoint(networkState.selectedAPIEndpoint, networkState.currentNetworkProfile?.httpPort));
-const localNetworkType = computed(() => ChainUtils.getNetworkType(networkState.currentNetworkProfile?.network.type));
+const localNetworkType = computed(() => AppState.networkType);
```
Comment (sleepyOwl14) [inline src/util/transactionUtils.ts:73]: no need this line of code

File: src/util/namespaceUtils.ts [hunk 6]
```diff
@@ -1,33 +1,23 @@
...
-import { WalletAccount } from "@/models/walletAccount";
-import { AutoAnnounceSignedTransaction, HashAnnounceBlock, AnnounceType, listenerState } from "@/state/listenerState";
-import { ListenerStateUtils } from "@/state/utils/listenerStateUtils";
+import {  listenerState } from "@/state/listenerState";
 import { ChainProfileConfig } from "@/models/stores/chainProfileConfig";
 import { AppState } from "@/state/appState";
+import { TransactionUtils } from "./transactionUtils";
 
+const networkType = AppState.networkType
+const generationHash = networkState.currentNetworkProfile.generationHash
```
Comment (sleepyOwl14) [inline src/util/namespaceUtils.ts:20]: this will result in a fix value, not dynamic

File: src/util/accountUtils.ts [hunk 7]
```diff
@@ -1,21 +1,16 @@
...
-import { ChainAPICall } from "@/models/REST/chainAPICall"
 import { WalletAccount } from "@/models/walletAccount";
 import { OtherAccount } from "@/models/otherAccount";
 import { Namespace } from "@/models/namespace";
 import { AppState } from "@/state/appState";
+import { TransactionUtils } from "./transactionUtils";
 
-const networkType = networkState.currentNetworkProfile.network.type;
+const networkType = AppState.networkType;
 const Hash = networkState.currentNetworkProfile.generationHash;
```
Comment (sleepyOwl14) [inline src/util/accountUtils.ts:14]: will not dynamic update value

File: src/util/multiSignatory.ts [hunk 8]
```diff
@@ -3,42 +3,23 @@ import {
...
-import { listenerState, AutoAnnounceSignedTransaction, HashAnnounceBlock, AnnounceType } from "@/state/listenerState";
 import { AppState } from "@/state/appState";
 import { OtherAccount } from "@/models/otherAccount";
-import { BuildTransactions } from "./buildTransactions";
 import { Helper } from "./typeHelper";
+import { ChainUtils } from "./chainUtils";
 
-const walletKey = "sw";
 const generationHash = networkState.currentNetworkProfile.generationHash
 const networkType = AppState.networkType
```
Comment (sleepyOwl14) [inline src/util/multiSignatory.ts:24]: not dynamic

File: src/util/buildTransactions.ts [hunk 9]
```diff
@@ -291,8 +291,4 @@ export class BuildTransactions {
     getFeeStrategy(): FeeCalculationStrategy {
         return this.transactionBuilderFactory.feeCalculationStrategy;
     }
-
-    setFeeStrategy(feeStrategy :FeeCalculationStrategy) :void{
-        this.transactionBuilderFactory.feeCalculationStrategy = feeStrategy; 
-    }
```
Comment (sleepyOwl14) [inline src/util/buildTransactions.ts:297]: this one can keep, no need remove

File: src/util/assetsUtils.ts [hunk 10]
```diff
@@ -2,59 +2,33 @@ import {
...
 import { AppState } from "@/state/appState";
+import { TransactionUtils } from "./transactionUtils";
 
 interface assetSelectionInterface {
   label: string,
   value: string,
 }
-
+const networkType = AppState.networkType;
+const generationHash = networkState.currentNetworkProfile.generationHash;
```
Comment (sleepyOwl14) [inline src/util/assetsUtils.ts:27]: not dynamic

**Review by sleepyOwl14 — CHANGES_REQUESTED** [review]
please fix accountUtil, assetUtil, namespaceUtil and multiSignatory networkType and generationHash

File: src/util/accountUtils.ts (hunk 3 above)
Comment (sleepyOwl14) [reply src/util/accountUtils.ts]: ya, will need to be able to select

Comment (anakin001) [reply src/util/accountUtils.ts]: ok, will update accordingly, thanks

**Review by sleepyOwl14 — APPROVED** [review]

**Pull request merged**